alicloud/resource_alicloud_vpn_route_entry.go

```diff
@@ -0,0 +1,219 @@
+package alicloud
+
+import (
+	"github.com/aliyun/alibaba-cloud-sdk-go/sdk/requests"
+	"github.com/aliyun/alibaba-cloud-sdk-go/services/vpc"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-alicloud/alicloud/connectivity"
+	"time"
+)
+
+func resourceAliyunVpnRouteEntry() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAliyunVpnRouteEntryCreate,
+		Read:   resourceAliyunVpnRouteEntryRead,
+		Update: resourceAliyunVpnRouteEntryUpdate,
+		Delete: resourceAliyunVpnRouteEntryDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"vpn_gateway_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"next_hop": {
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Required: true,
+			},
+
+			"route_dest": {
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Required: true,
+			},
+
+			"weight": {
+				Type:         schema.TypeInt,
+				Required:     true,
+				ValidateFunc: validateVpnBandwidth([]int{0, 100}),
+			},
+
+			"publish_vpc": {
+				Type:     schema.TypeBool,
+				Required: true,
+			},
+
+			// Computed values
+			"create_time": {
+				Type:     schema.TypeFloat,
+				Computed: true,
+			},
+
+			"old_weight": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+
+			"state": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAliyunVpnRouteEntryCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+	vpnRouteEntryService := VpnGatewayService{client}
+	request := vpc.CreateCreateVpnRouteEntryRequest()
+	request.RegionId = client.RegionId
+	request.VpnGatewayId = d.Get("vpn_gateway_id").(string)
+	request.RouteDest = d.Get("route_dest").(string)
+	request.NextHop = d.Get("next_hop").(string)
+	request.Weight = requests.NewInteger(d.Get("weight").(int))
+	request.PublishVpc = requests.NewBoolean(d.Get("publish_vpc").(bool))
+	request.ClientToken = buildClientToken(request.GetActionName())
+
+	if err := vpnRouteEntryService.WaitForVpnGateway(d.Get("vpn_gateway_id").(string), Active, 2*DefaultTimeout); err != nil {
+		return WrapError(err)
+	}
+
+	raw, err := client.WithVpcClient(func(vpcClient *vpc.Client) (interface{}, error) {
+		return vpcClient.CreateVpnRouteEntry(request)
+	})
+
+	if err != nil {
+		return WrapErrorf(err, DefaultErrorMsg, "alicloud_vpn_route_entry", request.GetActionName(), AlibabaCloudSdkGoERROR)
+	}
+	addDebug(request.GetActionName(), raw, request.RpcRequest, request)
+	response, _ := raw.(*vpc.CreateVpnRouteEntryResponse)
+
+	id := response.VpnInstanceId + ":" + response.NextHop + response.RouteDest
```

Comment:
I think the id should be `id := response.VpnInstanceId + ":" + response.NextHop + ":" +response.RouteDest`